lib/regexp_parser/scanner/scanner.rl

```diff
@@ -62,9 +62,15 @@
   quantifier_possessive = '?+' | '*+' | '++';
   quantifier_mode       = '?'  | '+';
 
-  quantifier_interval   = range_open . (digit+)? . ','? . (digit+)? .
+  quantifier_exact      = range_open . (digit+) . range_close . quantifier_mode?;
+  quantifier_minimum    = range_open . (digit+) . ',' . range_close . quantifier_mode?;
+  quantifier_maximum    = range_open . ',' . (digit+) . range_close . quantifier_mode?;
+  quantifier_range      = range_open . (digit+) . ',' . (digit+) .
                           range_close . quantifier_mode?;
 
+  quantifier_interval   = quantifier_exact | quantifier_minimum |
+                          quantifier_maximum | quantifier_range;
+
```

Comment:
slightly more concise:

```suggestion
  quantity_exact      = (digit+);
  quantity_minimum    = (digit+) . ',';
  quantity_maximum    = ',' . (digit+);
  quantity_range      = (digit+) . ',' . (digit+);

  quantifier_interval   = range_open . ( quantity_exact | quantity_minimum |
                          quantity_maximum | quantity_range ) . range_close .
                          quantitfier_mode?;

```

but this is more a matter of taste, ragel probably optimises away the redundancy (?)

Reply:
I like it a lot. The intention is clearer. There's a typo in the name of the mode pattern so I will make the change manually.